h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -454,12 +563,16 @@ private long copyCSVStream(Frame.CSVStream is, OutputStream os, int firstChkIdx)
     }
 
     private void exportCSVStream(Frame.CSVStream is, String path, int firstChkIdx) {
+      exportCSVStream(is, path, firstChkIdx, _compressor);
+    }
+
+    private void exportCSVStream(Frame.CSVStream is, String path, int firstChkIdx, CompressionFactory compressor) {
```

Comment:
Using MemoryMappedFiles of `java.nio` throughout the code could potentially boost the performance a lot: http://web.archive.org/web/20120821114802/http://geekomatic.ch/2009/01.html

http://web.archive.org/web/20120815094827/http://geekomatic.ch/2008/09.html

(not sure how these work on Windows, but few people care about windows).

would be a big change, yet mentioning it.

Reply:
Very interesting idea, thank you @Pscheidl. I will try it and report back the results!